src/avif.c

```diff
@@ -179,44 +180,43 @@ avifResult avifImageCopy(avifImage * dstImage, const avifImage * srcImage, avifP
     avifRWDataSet(&dstImage->exif, srcImage->exif.data, srcImage->exif.size);
     avifImageSetMetadataXMP(dstImage, srcImage->xmp.data, srcImage->xmp.size);
 
+    assert(AVIF_CHAN_A > AVIF_CHAN_V);
```

Comment:
Optional: This is probably a matter of preference. Although the original code is longer, it is more direct and easier to understand.

The new code splits the allocation of the plane buffers and the data copying, and the for loop that copies the data contains a special case for the YUV planes.

Reply:
> The new code splits the allocation of the plane buffers and the data copying

I find it more logical to allocate all the planes first, to avoid copying anything if the allocation failed.

> the for loop that copies the data contains a special case for the YUV planes.

It is actually a special case for UV only, so doing it during the YUV or the YUVA loop does not change much the issue.